include/aspect/postprocess/dynamic_topography.h

```diff
@@ -47,42 +47,79 @@ namespace aspect
         std::pair<std::string,std::string>
         execute (TableHandler &statistics);
 
+        /**
+         * Return the topography vector as calculated by CBF formulation
+         */
+        const LinearAlgebra::BlockVector &
+        topography_vector() const;
 
         /**
-         * Declare the parameters this class takes through input files.
+         * Return the cellwise topography vector as calculated by CBF formulation
          */
-        static
-        void
-        declare_parameters (ParameterHandler &prm);
+        const Vector<float> &
+        cellwise_topography() const;
 
         /**
-         * Read the parameters this class declares from the parameter file.
+         * Register the other postprocessor that we need: BoundaryPressures
          */
         virtual
+        std::list<std::string>
+        required_other_postprocessors() const;
+
+        /**
+         * Parse the parameters for the postprocessor.
+         */
         void
         parse_parameters (ParameterHandler &prm);
 
+        /**
+         * Declare the parameters for the postprocessor.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
       private:
         /**
-         * A parameter that we read from the input file that denotes whether
-         * we should subtract the mean topography or not.
+         * Output the dynamic topography solution to
+         * a file.
+         */
+        void output_to_file(bool upper, std::vector<std::pair<Point<dim>, double> > &values);
+
+        /**
+         * A vector which stores the surface stress values calculated
+         * by the postprocessor.
          */
-        bool subtract_mean_dyn_topography;
+        LinearAlgebra::BlockVector topo_vector;
 
         /**
-         * A parameter that specifies whether the lower topography is outputted.
+         * A vector which stores the surface stress values calculated
+         * at the midpoint of each surface cell face. This can be
+         * given to a visualization postprocessor to output dynamic topography.
          */
-        bool output_lower_surface_topography;
+        Vector<float> visualization_values;
 
         /**
-         * A parameter allows users to set the density value outside the surface
+         * A parameter allows users to set the density value
+         * above the top surface.
          */
         double density_above;
 
         /**
-         * A parameter allows users to set the density value below the surface
+         * A parameter allows users to set the density value
```

Comment:
Same here. Not your fault, but lets get it fixed, it was nearly copy-pasted into the geoid postprocessor.